### DISCUSSION TITLE
Toggle colors in ray app

### DISCUSSION THREAD

**Discussion by Polfo** [spatie/ray#392] [Ideas]
When I want to disable one color, I have to carefully click the 5 other colors in the ray app's top bar.
Wouldn't it be nice to have a toggle color filters button in the top bar? 

![image](https://user-images.githubusercontent.com/10303392/112472428-67f51a00-8d6d-11eb-81e8-90b4714de008.png)

**Comment by AdrianMrn** [comment]
What do you think about this: holding shift when clicking a color toggles all the other colors instead. This wouldn't be very obvious until you accidentally tried it, though.

@willemvb any ideas?

**Reply by Polfo** [reply]
Thank you for your suggestion, but are there other interfaces that work like that? There seems to be enough room for a button similar to this 'invert selection' button: https://iconscout.com/icon/invert-selection
![image](https://user-images.githubusercontent.com/10303392/112821980-a3eaf080-9087-11eb-9aee-dab13b2a24b6.png)

**Reply by willemvb** [reply]
Actually there isn't much room left when working with a small Ray window on MacOS.

<img width="425" alt="image" src="https://user-images.githubusercontent.com/1336390/114178659-50f22280-993e-11eb-9986-dbc30db50351.png">

I would be happy with a modifier key.

A pattern from iCal (calendar selection)
- Command+click on a deselected color: select all colors
- Command+click on a selected color: deselect all colors

A pattern from Photoshop (layer selection):
- Option+click on a deselected color: select only this color, and hide the rest
- Command+click on a selected color: deselect this color, show all others

**Comment by aryehraber** [comment]
I was about to request this feature too, would love to see it added!

I think the above proposed solution by holding a modifier key + clicking a color would be perfect, in fact I intuitively tried `option+click` before coming here 😅
